### PR TITLE
feat(hyd): Low speed pump efficiency and refactor cargo door for all planes

### PR DESCRIPTION
**Pull request by crocket63**
Fixes #[issue_no]

## Summary of Changes

Added an efficiency threshold so hydraulic pumps cannot properly work at extremely slow rotational speeds.

Small refactoring of cargo door code so we remove A320/A380 code duplication.

## Additional context

Discord username (if different from GitHub):

## Testing instructions

- Non regression on global hydraulics
- Check cargo door operation is still correct

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page

### CODE REVIEW

File: fbw-common/src/wasm/systems/systems/src/hydraulic/cargo_doors.rs [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+        self.update_timers(context);
+        self.update_actions_from_state();
+    }
+
+    fn update_timers(&mut self, context: &UpdateContext) {
+        if self.control_state == DoorControlState::NoControl {
+            self.duration_in_no_control += context.delta();
+        } else {
+            self.duration_in_no_control = Duration::from_secs(0);
+        }
```
Comment (2hwk) [inline fbw-common/src/wasm/systems/systems/src/hydraulic/cargo_doors.rs:95]: Could use a simple assignment here, might look cleaner... but also looks fine as is

File: fbw-common/src/wasm/systems/systems/src/hydraulic/cargo_doors.rs [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+
+        if self.control_state == DoorControlState::HydControl {
+            self.duration_in_hyd_control += context.delta();
+        } else {
+            self.duration_in_hyd_control = Duration::from_secs(0);
+        }
+    }
```
Comment (2hwk) [inline fbw-common/src/wasm/systems/systems/src/hydraulic/cargo_doors.rs:102]: Ditto

File: fbw-common/src/wasm/systems/systems/src/hydraulic/cargo_doors.rs (hunk 1 above)
Comment (crocket63) [reply fbw-common/src/wasm/systems/systems/src/hydraulic/cargo_doors.rs:95]: What do you mean exactly?

**Comment by mico975** [comment]
Quality Assurance Tester Report

Discord          : mico#3145
Object of testing: #7926
Tier of Testing  : 1
Date             : 03/04/2023

Testing Process: 
Non regression on global hydraulics - **OK**
Check cargo door operation is still correct - **OK**

Negatives:       
N/A

Testing Results: 
Passed

Conclusions: 
LGTM

File: fbw-common/src/wasm/systems/systems/src/hydraulic/mod.rs [hunk 3]
```diff
@@ -2623,10 +2626,15 @@ impl Pump {
         }
     }
 
-    fn calculate_flow(speed: AngularVelocity, displacement: Volume) -> VolumeRate {
-        if speed.get::<revolution_per_minute>() > 0. {
+    fn get_max_flow_from_max_dsiplacement(&self) -> VolumeRate {
```
Comment (BlueberryKing) [inline fbw-common/src/wasm/systems/systems/src/hydraulic/mod.rs:2629]:
```suggestion
    fn get_max_flow_from_max_displacement(&self) -> VolumeRate {
```

**Review by BlueberryKing — CHANGES_REQUESTED** [review]

File: fbw-common/src/wasm/systems/systems/src/hydraulic/mod.rs (hunk 3 above)
Comment (crocket63) [reply fbw-common/src/wasm/systems/systems/src/hydraulic/mod.rs:2629]: :(

**Pull request merged**